### PR TITLE
Update documentation for month constants

### PR DESCRIPTION
**Pull request by eulerkochy**
Addresses a part of #31202

### CODE REVIEW

File: stdlib/Dates/docs/src/index.md [hunk 1]
```diff
@@ -710,6 +710,29 @@ Dates.dayofweek
...
+Dates.August
+Dates.Aug
+Dates.September
+Dates.Sep
+Dates.October
+Dates.Oct
+Dates.November
+Dates.Nov
+Dates.December
+Dates.Dec
```
Comment (fredrikekre) [inline stdlib/Dates/docs/src/index.md]: These are all documented in a nice table below. I suggest leaving out these ones from the manual, and just keep the table.

Comment (eulerkochy) [reply stdlib/Dates/docs/src/index.md]: Onto it!

Comment (eulerkochy) [reply stdlib/Dates/docs/src/index.md]: So, I delete this entirely, right?

Comment (fredrikekre) [reply stdlib/Dates/docs/src/index.md]: Yea.

Comment (eulerkochy) [reply stdlib/Dates/docs/src/index.md]: Done!

**Review by cormullion — COMMENTED** [review]
"abrreviation" -> "abbreviation"

**Comment by StefanKarpinski** [comment]
Documenting each month individually is kind of amusing but we do want to document everything, so 👍

**Comment by eulerkochy** [comment]
I wanted to opt for metaprogramming but there were far too many cases to consider for each month, so I adopted this simple, yet exhaustive method!

**Comment by StefanKarpinski** [comment]
That works. We're unlikely to get any more months any time soon.

File: stdlib/Dates/src/query.jl [hunk 2]
```diff
@@ -226,9 +226,304 @@ function daysofweekinmonth(dt::TimeType)
...
+
+Abbreviation for [`November`](@ref).
+
+# Examples
+```jldoctest
+julia> Nov
+11
+```
+"""
+const Nov= 11
```
Comment (StefanKarpinski) [inline stdlib/Dates/src/query.jl]:
```suggestion
const Nov = 11
```

**Comment by StefanKarpinski** [comment]
I'm a little surprised that these are just integer constants and not some kind of `Month` object but I that's how it was before this PR so documenting them doesn't make anything worse.

**Comment by StefanKarpinski** [comment]
@quinnj, does this look good to you?

**Comment by eulerkochy** [comment]
Bump

**Pull request merged**